tests/transformers/test_transformers_model_export.py

```diff
@@ -898,3 +1043,645 @@ def postprocess(self, model_outputs):
+    signature = infer_signature(data, infer_pyfunc.predict(data))
+
+    mlflow.transformers.save_model(pipeline, model_path, signature=signature)
+    loaded_pyfunc = mlflow.pyfunc.load_model(model_path)
+    inference = loaded_pyfunc.predict(data)
+
+    assert inference == result
+
+    if len(data) > 1 and isinstance(data, list):
+        pd_input = pd.DataFrame([{"inputs": v} for v in data])
+    elif isinstance(data, list) and len(data) == 1:
+        pd_input = pd.DataFrame([{"inputs": v} for v in data], index=[0])
+    else:
+        pd_input = pd.DataFrame({"inputs": data}, index=[0])
+
+    pd_inference = loaded_pyfunc.predict(pd_input)
+    assert pd_inference == result
+
+
+@pytest.mark.parametrize(
+    "pipeline_name, example, in_signature, out_signature",
+    [
+        (
+            "fill_mask_pipeline",
+            ["I use stacks of <mask> to buy things", "I <mask> the whole bowl of cherries"],
+            [{"type": "string"}],
+            [{"type": "string"}],
+        ),
+        (
+            "zero_shot_pipeline",
+            {
+                "sequences": ["My dog loves to eat spaghetti", "My dog hates going to the vet"],
+                "candidate_labels": ["happy", "sad"],
+                "hypothesis_template": "This example talks about how the dog is {}",
+            },
+            [
+                {"name": "sequences", "type": "string"},
+                {"name": "candidate_labels", "type": "string"},
+                {"name": "hypothesis_template", "type": "string"},
+            ],
+            [{"type": "string"}],
+        ),
+    ],
+)
+@pytest.mark.parametrize("provide_example", [True, False])
+def test_infer_signature_from_example_only(
+    pipeline_name, model_path, example, request, provide_example, in_signature, out_signature
+):
+    pipeline = request.getfixturevalue(pipeline_name)
+
+    input_example = example if provide_example else None
+    mlflow.transformers.save_model(pipeline, model_path, input_example=input_example)
+
+    model = Model.load(model_path)
+
+    assert model.signature.inputs.to_dict() == in_signature
+    assert model.signature.outputs.to_dict() == out_signature
+
+    if provide_example:
+        saved_example = _read_example(model, model_path).to_dict(orient="records")
+        if isinstance(example, str):
+            assert next(iter(saved_example[0].values())) == example
+        elif isinstance(example, list):
+            assert list(saved_example[0].values()) == example
+        else:
+            assert set(saved_example[0].keys()).intersection(example.keys()) == set(
+                saved_example[0].keys()
+            )
+    else:
+        assert model.saved_input_example_info is None
+    # NB: do not purge the cache during this test. The pipeline will attempt to refer to
+    # the cache state within the fixture context which will throw a file not found error
+    # when attempting to read the stored FastTokenizer for the TextClassificationPipeline.
```

Comment:
Is this because `scope` was set to `"module"`?

Reply:
indeed. Removed this note and only disabled the cache purge for two tests that exhibited flakiness when I was locally running. There was an attempt to do a cache fetch that popped up 2/20 times but only for those tests.